### DISCUSSION TITLE
How to pass data from one query to another when props are required but query returns undefined/data

### DISCUSSION THREAD

**Discussion by stychu** [TanStack/query#6869] [Q&A] [answered]
I have a mutation which then is used for query. The query uses the `id` that comes from mutation and its required. How do I handle typescript yelling at  
`Type  undefined  is not assignable to type  string `

```ts
 const downloadMutation = api.batch.triggerDownload.useMutation();
  const log = api.batch.requestLog.useQuery(
    { id: downloadMutation.data?.jobId },  // Type  undefined  is not assignable to type  string 
    { enabled: !!downloadMutation?.data },
  );
  ```

**Comment by TkDodo** [comment, marked as answer]
https://tkdodo.eu/blog/react-query-and-type-script#type-safety-with-the-enabled-option
